flow/projecttracking/jira/jira.py

```diff
@@ -0,0 +1,376 @@
+        # else calculate_next_semver will throw an error about getting 'None'
+        if bump_type is None:
+            bump_type = 'minor'
+
+        commons.print_msg(Jira.clazz, method, "bump type: {}".format(bump_type))
+
+        commons.print_msg(Jira.clazz, method, 'end')
+
+        return bump_type
+
+    def extract_story_id_from_commit_messages(self, commit_messages):
+        method = 'extract_story_id_from_commit_messages'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        story_list = []
+
+        for commit_string in commit_messages:
+            
+            # check if there is a starting bracket and if there are balanced brackets
+            if commit_string.count('[') > 0 and commit_string.count('[') == commit_string.count(']'):
+                # for each starting bracket
+                for m in re.finditer('\[', commit_string):
+                    # find the next subsequent ending bracket
+                    ending_bracket = commit_string.find(']', m.start())
+                    # find the contents between the brackets
+                    stories = commit_string[m.start()+1:ending_bracket]
+
+                    # verify there isn't a embedded bracket, if so just skip this one and keep marching.
+                    if stories.find('[') == -1:  # there is a nested starting bracket
+                        # now dig out the tracker number or jira key in single number format or multiple separated by commas.
+                        r = re.compile('(?:[a-zA-Z]+\-[0-9]+,?)+(,([a-zA-Z]+\-[0-9]+,?))*,?')
+                        stories_array = stories.split(',')
+                        stories = list(filter(r.match, stories_array))
+                        for story in stories:
+                            # split out by comma.
+                            if story not in story_list:
+                                story_list.append(story)
+
+        commons.print_msg(Jira.clazz, method, "Story list: {}".format(story_list))
+        commons.print_msg(Jira.clazz, method, 'end')
+        return story_list
+
+    """
+        This methods needs to return an array of stories with 4 specific fields for each story:
+            story_type
+            id
+            name
+            description
+            url
+            current_state
+    """
+    def flatten_story_details(self, story_details):
+        method = 'flatten_story_details'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        if story_details is None:
+            return None
+        story_release_notes = []
+        for story in story_details:
+            story_release_note_summary = {}
+            story_release_note_summary['story_type'] = story.get('fields').get('issuetype').get('name').lower()
+            story_release_note_summary['id'] = story.get('key').upper()
+            story_release_note_summary['name'] = story.get('fields').get('summary')
+            story_release_note_summary['url'] = story.get('self')
```

Comment:
The URL for each JIRA item that gets populated to the github release notes is the URL to the jira API endpoint which is not what we probably want.  For example:

![image](https://user-images.githubusercontent.com/6393612/117983819-4c39d900-b305-11eb-9c98-90b0ed782c87.png)

... has the following URLs for the jira items:

* https://redacted.atlassian.net/rest/api/3/issue/11434
* https://redacted.atlassian.net/rest/api/3/issue/11322

Which results in a page like this:
![image](https://user-images.githubusercontent.com/6393612/117984392-cff3c580-b305-11eb-883e-40e7728c0d6f.png)


When in fact they should be,

* https://redacted.atlassian.net/browse/JIRA-1035
* https://redacted.atlassian.net/browse/JIRA-1034

Which results in a page like this:
![image](https://user-images.githubusercontent.com/6393612/117984518-e8fc7680-b305-11eb-904b-564f8c0c8280.png)


We probably need to retrieve or construct a uset-friendly URL for use in the release notes payload buildout.

Reply:
Done